bionic/provider.py

```diff
@@ -99,6 +99,18 @@ def entity_names(self):
             name for dnode in self.attrs.dnodes for name in dnode.all_entity_names()
         ]
 
+    # This is used to indicate that if we're outputting an aggregate descriptor type
+    # (like a tuple), then we'll need to set some followup tasks to make sure all the
+    # aggregated descriptors get persisted immediately. This field is gross, but once we
+    # have file descriptors it can go away, because the "crude" or "not-yet-persisted"
+    # values will have their own descriptors, and it will be obvious which tasks need
+    # to have followups.
+    # TODO In the meantime, if we start moving more attributes directly onto the Task
+    # class, this would be another good candidate.
+    @property
+    def task_output_may_need_followups_for_persistence(self):
+        return True
```

Comment:
I don't understand what providers would use this property to be true? Is there an easy way where we can figure out those providers and set it to `True` only for them, while keeping the default value `False`?

Reply:
We discussed this in a call, but to summarize: this should be true for any provider than can provide a tuple, except `TupleConstructionProvider`.

In a sense, the underlying problem here is that tuples are sometimes upstream of their components and sometimes downstream, which creates a bunch of logical issues, including this one. Once we have file descriptors that will be resolved: crude tuples will always be upstream; regular tuples will always be downstream.